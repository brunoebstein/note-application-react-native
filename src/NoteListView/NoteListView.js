import React from 'react'
import {
	StyleSheet,
	Text,
	View,
	Button,
} from 'react-native';


export const NoteListView = () => (
	<View style={styles.container}>
		<Text style={styles.welcome}>
			Vous n'avez pas de notes
		</Text>
		<View style={styles.addNoteButton}>
			<Button
				onPress={() => null}
				title="Create"
			/>
		</View>
	</View>
)


const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#F5FCFF',
	},
	welcome: {
		fontSize: 20,
		textAlign: 'center',
		margin: 10,
	},
	addNoteButton: {
		position: 'absolute',
		bottom: 20,
		right: 20,
	},
});
