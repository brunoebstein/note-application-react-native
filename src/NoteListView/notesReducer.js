import { SAVE_NOTE } from '../actionTypes';

const defaultState = [];

export const notesReducer = (state = defaultState, action) => {
  switch (action && action.type) {
    case SAVE_NOTE:
      return state.concat(action.payload);
    default:
      return state;
  }
};
