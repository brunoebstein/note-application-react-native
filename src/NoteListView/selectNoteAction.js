import { SELECT_NOTE } from '../actionTypes';

export const selectNote = note => ({
  type: SELECT_NOTE,
  payload: note,
});
