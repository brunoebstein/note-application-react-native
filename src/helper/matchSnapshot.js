import { shallow } from './shallow';

export const matchSnapshot = component => {
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
};
