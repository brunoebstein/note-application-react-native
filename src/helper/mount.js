import './enzyme';

export { mount } from 'enzyme';

import { JSDOM } from 'jsdom';

// @TODO: update global just for the current test
global.window = new JSDOM('').window;
Object.keys(window).forEach(property => {
  if (typeof global[property] === 'undefined') {
    global[property] = window[property];
  }
});
