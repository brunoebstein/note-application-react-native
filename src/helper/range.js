export function range(max) {
  // return [...Array(20).keys()]; // doesn't work the same way in iOS & Android : broken on Android
  return Array.from({ length: max }, (_, i) => i);
}
