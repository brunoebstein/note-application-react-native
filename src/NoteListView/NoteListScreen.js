import React, { Component } from 'react';

import { themedNavigationOptions } from '../NoteNavigator/themedNavigationOptions';
import { NoteListViewWired } from '../NoteListView/NoteListViewWired';

export class NoteListScreen extends Component {
  static navigationOptions = themedNavigationOptions({
    title: 'Notes',
  });

  render() {
    const {
      navigation: { navigate },
    } = this.props;
    return (
      <NoteListViewWired onSelectedNote={() => navigate('NoteEditorView')} />
    );
  }
}
