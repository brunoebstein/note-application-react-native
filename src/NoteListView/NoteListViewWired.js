import { connect } from 'react-redux';

import { notesSelector } from './notesSelector';
import { selectNote } from './selectNoteAction.js';
import { NoteListView } from './NoteListView';

const mapStateToProps = state => ({
  notes: notesSelector(state),
});

export const NoteListViewWired = connect(
  mapStateToProps,
  { selectNote },
)(NoteListView);
