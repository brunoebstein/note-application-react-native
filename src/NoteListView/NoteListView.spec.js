import { ThemeProvider } from 'styled-components';

import { range } from '../helper/range';
import { matchSnapshot } from '../helper/matchSnapshot';

import { NoteListView } from './NoteListView';

describe('NoteListView component', () => {
  function matchSnapshotWithProvider(element) {
    matchSnapshot(<ThemeProvider theme={{}}>{element}</ThemeProvider>);
  }

  it('should display an empty list when no note', () => {
    matchSnapshotWithProvider(<NoteListView />);
  });

  it('should display title note when 1 note', () => {
    const notes = [{ title: 'my title' }];
    matchSnapshotWithProvider(<NoteListView notes={notes} />);
  });

  it('should display title note when 2 notes', () => {
    const notes = [{ title: 'my title' }, { title: 'my title 2' }];
    matchSnapshotWithProvider(<NoteListView notes={notes} />);
  });

  it('should display title note when many notes', () => {
    const notes = range(20).map(i => ({ title: `my title ${i}` }));
    matchSnapshotWithProvider(<NoteListView notes={notes} />);
  });
});
