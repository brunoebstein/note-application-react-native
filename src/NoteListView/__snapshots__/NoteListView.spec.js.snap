// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteListView component should display an empty list when no note 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 20,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Vous n'avez pas de notes
  </Text>
  <View
    style={
      Array [
        Object {
          "bottom": 20,
          "position": "absolute",
          "right": 20,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Create
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`NoteListView component should display title note when 1 note 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView>
    <View>
      <RCTScrollView
        data={
          Array [
            Object {
              "title": "my title",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "bottom": 20,
          "position": "absolute",
          "right": 20,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Create
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`NoteListView component should display title note when 2 notes 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView>
    <View>
      <RCTScrollView
        data={
          Array [
            Object {
              "title": "my title",
            },
            Object {
              "title": "my title 2",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 2
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "bottom": 20,
          "position": "absolute",
          "right": 20,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Create
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`NoteListView component should display title note when many notes 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView>
    <View>
      <RCTScrollView
        data={
          Array [
            Object {
              "title": "my title 0",
            },
            Object {
              "title": "my title 1",
            },
            Object {
              "title": "my title 2",
            },
            Object {
              "title": "my title 3",
            },
            Object {
              "title": "my title 4",
            },
            Object {
              "title": "my title 5",
            },
            Object {
              "title": "my title 6",
            },
            Object {
              "title": "my title 7",
            },
            Object {
              "title": "my title 8",
            },
            Object {
              "title": "my title 9",
            },
            Object {
              "title": "my title 10",
            },
            Object {
              "title": "my title 11",
            },
            Object {
              "title": "my title 12",
            },
            Object {
              "title": "my title 13",
            },
            Object {
              "title": "my title 14",
            },
            Object {
              "title": "my title 15",
            },
            Object {
              "title": "my title 16",
            },
            Object {
              "title": "my title 17",
            },
            Object {
              "title": "my title 18",
            },
            Object {
              "title": "my title 19",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 0
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 1
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 2
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 3
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 4
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 5
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 6
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 7
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 8
            </Text>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "color": "black",
                    "fontSize": 26,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              my title 9
            </Text>
          </View>
          <View
            style={
              Object {
                "height": 0,
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "bottom": 20,
          "position": "absolute",
          "right": 20,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Create
        </Text>
      </View>
    </View>
  </View>
</View>
`;
